components/background_hang_monitor/background_hang_monitor.rs

```diff
@@ -112,6 +125,7 @@ impl BackgroundHangMonitorChan {
             sender,
             component_id: component_id,
             disconnected: Default::default(),
+            monitoring_enabled: opts::get().background_hang_monitor,
```

Comment:
I would prefer to add new arguments that are provided by callers instead of adding more uses of opts::get(). One benefit is that it makes writing unit tests easier.